### PR TITLE
Update provider docs to use new directory format

### PR DESCRIPTION
**Pull request by davidji99**
New docs directory moves from `website/` to `docs/`. Filenames drop the `.html.markdown` reference and just use `.md`.

Additionally, I'm updating the doc themselves by checking for any syntax changes, broken links, etc.

### CODE REVIEW

**Review by bernerdschaefer — APPROVED** [review]
Oh, wonderful! I mostly skimmed, so let me know if there are any specific changes you want extra eyes on. Otherwise looks good to me! :shipit:

File: README.md [hunk 1]
```diff
@@ -13,14 +13,14 @@ Requirements
...
 
 Usage
 -----
 
 For Terraform 0.12+ compatibility, the configuration should specifiy version 2 or higher:
 
-```hcl
+```hcl-terraform
 provider "heroku" {
   version = "~> 2.0"
```
Comment (mars) [inline README.md:25]: This example version should probably updated for the current major version.

File: docs/resources/app.md [hunk 2]
```diff
@@ -48,34 +48,44 @@ The following arguments are supported:
 
 * `name` - (Required) The name of the application. In Heroku, this is also the
    unique ID, so it must be unique and have a minimum of 3 characters.
+
```
Comment (mars) [inline docs/resources/app.md:51]: These line-breaks between Markdown bullet list items actually makes every item a separate list.

Please remove these added inter-item line breaks.

File: docs/resources/formation.md [hunk 3]
```diff
@@ -13,16 +13,16 @@ resource.
 
 A formation represents the formation of processes that should be set for an application.
 
-~> **NOTE:** 
-- The application must have a dyno in order to update its formation.
-- If the heroku formation resource is removed and deleted, this will be a no-op action in Heroku.
+Please note the following:
```
Comment (mars) [inline docs/resources/formation.md:16]: Thank you for fixing these Hashicorp-specific style anomalies!

**Review by mars — CHANGES_REQUESTED** [review]
Nice improvements in addition to the fundamental directory fix 🙌😄

File: README.md (hunk 1 above)
Comment (davidji99) [reply README.md:25]: Done.

File: docs/resources/app.md (hunk 2 above)
Comment (davidji99) [reply docs/resources/app.md]: Changed.

**Review by mars — APPROVED** [review]

**Pull request merged**